keps/sig-network/2091-admin-network-policy/README.md

```diff
@@ -408,164 +398,186 @@ also like to consider the following set of proposals as future work items:
-	// source/dest). If this field is present and contains at least one item, this rule
-	// allows/denies traffic only if the traffic matches at least one item in the from/to list.
-	// +optional
-	From/To      []ClusterNetworkPolicyPeer
-	// Name is an identifier for this rule. Must be unique within an instance of CNP. Can be
-	// same across policy instances.
-  // +optional
-	Name         string
+// Exactly one of the selector pointers should be set in a given AppliedTo
+type AppliedTo struct {
+  NamespaceSelector         *metav1.LabelSelector
+  NamespaceAndPodSelector   *NamespaceAndPodSelector
 }
 
-// WorkloadSelector selects workloads based on the label Selector.
-// Right now the only selector type supported is PodSelector. More selector types may be
-// added in the future to support other workload selection mechanisms.
-// Must be accompanied by a NamespaceSelector to specify the scope of the workload selection.
-type WorkloadSelector struct {
-	PodSelector         *metav1.LabelSelector
+// NamespaceAndPodSelector allows the user to select a namespace/s and optionally 
+// select a given set of pods in that namespace/s
+type NamespaceAndPodSelector struct {
+  NamespaceSelector   metav1.LabelSelector
+  // Used to explicitly select pods within a namespace
+  PodSelector         *metav1.LabelSelector
 }
-```
 
-**Alternatives considered for the WorkloadSelector struct:**  
-```golang
-type WorkloadSelector struct {
-	selectorType        WorkloadSelectorType
-	selector            metav1.LabelSelector
+type AdminNetworkPolicyIngressRule struct {
+  // Name is an identifier for this rule. Must be unique within an instance of ANP. Can be
+  // same across policy instances.
+  // +optional
+  Name      string
+  // Action specifies whether this rule must pass, allow or deny traffic. 
+  // Allow: explicitly allows the selected traffic 
+  // Deny: explicitly denies the selected traffic
+  // Pass: explicitly allows the selected traffic to skip and remaining ANP rules 
+  // and be delegated by K8's Network Policy. 
+  Action    RuleAction
+  // List of ports for incoming traffic.
+  // Each item in this list is combined using a logical OR. If this field is
+  // empty or missing, this rule matches no ports (traffic not restricted by port).
+  // If this field is present and contains at least one item, then this rule 
+  // allows/denies/passes traffic only if the traffic matches at least one port in the list.
+  // +optional
+  Ports     []networkingv1.NetworkPolicyPort
+	// List of sources from which traffic will be allowed/denied/passed to the entities 
+  // selected by this AdminNetworkPolicyRule. Items in this list are combined using a logical OR 
+  // operation. If this field is empty or missing, this rule matches no sources. 
+  // If this field is present and contains at least one item, this rule
+	// allows/denies/passes traffic from the defined AdminNetworkPolicyPeer
+	From      []AdminNetworkPolicyPeer
```

Comment:
Indent

Reply:
Weird it renders correctly in my vscode :), fixed